Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/detail/AgendaDetailViewModel.kt

```diff
@@ -0,0 +1,102 @@
+package com.mentalmachines.droidcon_boston.views.detail
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.mentalmachines.droidcon_boston.R
+import com.mentalmachines.droidcon_boston.data.FirebaseDatabase
+import com.mentalmachines.droidcon_boston.data.Schedule
+import com.mentalmachines.droidcon_boston.data.UserAgendaRepo
+import com.mentalmachines.droidcon_boston.firebase.FirebaseHelper
+import com.mentalmachines.droidcon_boston.utils.NotificationUtils
+
+class AgendaDetailViewModel(
+    private val scheduleRowItem: Schedule.ScheduleRow,
+    private val userAgendaRepo: UserAgendaRepo,
+    private val firebaseHelper: FirebaseHelper = FirebaseHelper.instance
+) : ViewModel() {
+    private val eventSpeakers: HashMap<String, FirebaseDatabase.EventSpeaker> = hashMapOf()
+
+    private val _scheduleDetail = MutableLiveData<Schedule.ScheduleDetail>()
+    val scheduleDetail: LiveData<Schedule.ScheduleDetail> = _scheduleDetail
+
+    private val dataListener: ValueEventListener = object : ValueEventListener {
+        override fun onCancelled(databaseError: DatabaseError) {
+            Log.e(javaClass.canonicalName, "detailQuery:onCancelled", databaseError.toException())
```

Comment:
Probably should add Timber?  I always wondered why we never used that...probably should have just added it last year...